pkg/domain/infra/abi/images.go

```diff
@@ -461,13 +461,21 @@ func (ir *ImageEngine) Search(ctx context.Context, term string, opts entities.Im
 
 	searchOptions := &libimage.SearchOptions{
 		Authfile:              opts.Authfile,
+		CertDirPath:           opts.CertDir,
+		Username:              opts.Username,
+		Password:              opts.Password,
 		Filter:                *filter,
 		Limit:                 opts.Limit,
 		NoTrunc:               true,
 		InsecureSkipTLSVerify: opts.SkipTLSVerify,
 		ListTags:              opts.ListTags,
 	}
 
+	// Only set IdentityToken if it is not empty.
+	if opts.IdentityToken != "" {
+		searchOptions.IdentityToken = opts.IdentityToken
```

Comment:
AFAICS this is equivalent in effect to unconditionally setting `IdentityToken: opts.IdentityToken` in the initializer above. — and consistent with how the other optional fields like `Username` and `CertDirPath` are handled.

Reply:
Changed to `IdentityToken: opts.IdentityToken` and CI test passed.